### PR TITLE
Add depth camera grabability diagnostics

### PR DESCRIPTION
**Pull request by brinij**
This PR adds diagnostics about depth image.
If there are no subscribers, then we do not know if depth image grabbing works or not, but we also do not care (publish OK in diagnostics)
But if there are subscribers, we compare the last timestamp with the current timestamp, if it exceeds the threshold (5.0 seconds are big enough number because the camera publishes with at least 10 Hz, and there is no need to have it as a parameter), then we throw a diagnostics error.

### CODE REVIEW

**Review by tkachuk-magazino — APPROVED** [review]

File: src/astra_driver.cpp [hunk 1]
```diff
@@ -165,6 +165,7 @@ AstraDriver::AstraDriver(ros::NodeHandle& n, ros::NodeHandle& pnh) :
   diagnostics_updater_.verbose_ = false;
   diagnostics_updater_.setHardwareID("astra_" + serial_number_);
   diagnostics_updater_.add("camera_availability", this, &AstraDriver::getAvailabilityDiagnostic);
+  diagnostics_updater_.add("depth_grabability", this, &AstraDriver::getDepthGrabability);
```
Comment (carzum) [inline src/astra_driver.cpp]: I dont think grabability is the right word here.  No body grabs the image. maybe just depth_availability?

File: src/astra_driver.cpp [hunk 2]
```diff
@@ -188,6 +189,7 @@ AstraDriver::AstraDriver(ros::NodeHandle& n, ros::NodeHandle& pnh) :
   depth_action_server_->start();
 
   diagnostics_timer_ = pnh_.createTimer(ros::Duration(1), &AstraDriver::diagnosticsTimerCallback, this);
+  last_depth_frame_timestamp_ = ros::Time::now();
```
Comment (carzum) [inline src/astra_driver.cpp]: maybe 0? now() could cause minor hickups and race conditions when the driver starts

File: src/astra_driver.cpp [hunk 3]
```diff
@@ -268,6 +270,21 @@ void AstraDriver::getAvailabilityDiagnostic(diagnostic_updater::DiagnosticStatus
   }
 }
 
+void AstraDriver::getDepthGrabability(diagnostic_updater::DiagnosticStatusWrapper &stat)
```
Comment (carzum) [inline src/astra_driver.cpp]: here i would remove the grabs

File: src/astra_driver.cpp [hunk 4]
```diff
@@ -268,6 +270,21 @@ void AstraDriver::getAvailabilityDiagnostic(diagnostic_updater::DiagnosticStatus
...
+{
+  if (depth_raw_subscribers_ || depth_subscribers_){
+    if ((ros::Time::now() - last_depth_frame_timestamp_) > ros::Duration(5.0)){
```
Comment (carzum) [inline src/astra_driver.cpp]: 5.0 should be constant

**Review by carzum — CHANGES_REQUESTED** [review]

**Review by domeinzer — APPROVED** [review]

**Review by carzum — APPROVED** [review]

**Pull request merged**